src/libstd/io/mod.rs

```diff
@@ -1376,6 +1377,69 @@ pub trait Write {
         Ok(())
     }
 
+    /// Attempts to write an multiple buffers into this writer.
+    ///
+    /// This method will continuously call [`write_vectored`] until there is no
+    /// more data to be written or an error of non-[`ErrorKind::Interrupted`]
+    /// kind is returned. This method will not return until all buffers have
+    /// been successfully written or such an error occurs. The first error that
+    /// is not of [`ErrorKind::Interrupted`] kind generated from this method
+    /// will be returned.
+    ///
+    /// If the buffer contains no data, this will never call [`write_vectored`].
+    ///
+    /// [`write_vectored`]: #method.write_vectored
+    /// [`ErrorKind::Interrupted`]: ../../std/io/enum.ErrorKind.html#variant.Interrupted
+    ///
+    /// # Notes
+    ///
+    /// Different to `io::Write::write_vectored` this takes a *mutable*
+    /// reference to a slice of `IoSlice`s, not a non-mutable reference, because
+    /// we need to modify the slice to keep track of the bytes already written.
+    ///
+    /// Once this function returns the contents of `bufs` is unspecified, as we
+    /// don't know what the contents of `bufs` will be as that depends on how
+    /// many writes we needed to do. We advice to see this function as taking
+    /// ownership of `bufs` and don't use the variable after the future returns.
+    /// The underlying buffers, to which `IoSlice` points (not the `IoSlice`
+    /// itself), are unchanged and can be reused.
```

Comment:
I think this paragraph could be improved. 

First sentence: "is" -> "are". And I would remove the middle part. Maybe something like this?

> Once this function returns, the contents of `bufs` are unspecified, as this depends on how many calls to `write_vectored` were necessary.

Second sentence: what future? Alternative suggestion:

> It is best to understand this function as taking ownership of `bufs` and to not use `bufs` afterwards.

Third sentence: `IoSlice` should be plural I guess (and I am not sure about the first comma...):

> The underlying buffers, to which the `IoSlice`s point (but not the `IoSlice`s themselves), are unchanged and can be reused.

Reply:
> First sentence: "is" -> "are". And I would remove the middle part. Maybe something like this?

Are you sure about this, isn't a single variable? I'l change it anyway.

> Second sentence: what future? Alternative suggestion:

I copied the docs I wrote for the `WriteAllVectored` `Future` in the futures crate :).